Sources/PackageRegistry/RegistryClient.swift

```diff
@@ -2441,3 +2441,35 @@ extension URLComponents {
         path += (path.last == "/" ? "" : "/") + components.joined(separator: "/")
     }
 }
+
+extension FileSystem {
+    /// Get the contents of a file without loading all of it in memory
+    ///
+    /// - Returns: The file contents as Data, or nil if missing
+    func readFileInChunks(path: AbsolutePath, bufferSize: Int = 1024) -> Data? {
```

Comment:
Sorry, I'm not sure I understand what this is trying to achieve. How exactly would this prevent reading the whole file into memory if the resulting `Data` value contains the whole file in memory by definition?

Reply:
You're totally right. 
My bad, this is actually far from being ready for review. 
Marking it as a draft for now if that's ok.